Commencement.Mvc/Controllers/EmailQueueController.cs

```diff
@@ -63,7 +63,10 @@ public ActionResult AllStudentEmail(string id)
                 throw new Exception("Id not supplied");
             }
 
-            var queue = _emailQueueRepository.Queryable.Where(a => a.Student.StudentId == id).ToArray();
+            var studentList = _studentRepository.Queryable.Where(a => a.StudentId == id).Select(s => s.Id).ToArray();
+
+
+            var queue = _emailQueueRepository.Queryable.Where(a => studentList.Contains(a.Student.Id)).ToArray();

```

Comment:
@jSylvestre why would the student list query return more than one studentID?  If you just get one it'll really simplify the queries